spring-thrift-starter/src/main/java/info/developerblog/spring/thrift/client/pool/ThriftClientPooledObjectFactory.java

```diff
@@ -74,7 +100,7 @@ public TServiceClient create(ThriftClientKey key) throws Exception {
     public void activateObject(ThriftClientKey key, PooledObject<TServiceClient> p) throws Exception {
         super.activateObject(key, p);
 
-        Optional requestId = Optional.fromNullable(MDC.get(requestIdLogger.getMDCKey()));
+        Optional requestId = Optional.ofNullable(MDC.get(requestIdLogger.getMDCKey()));
```

Comment:
If you don't lock on Java version, and if you are using SpringBoot you don't lock (yep :)), you should choosing highest stable version of Java. I think about downgrade, but I don't see any reason.